extensions/extended-isupport.md

```diff
@@ -0,0 +1,92 @@
+---
+title: "`extended-isupport` Extension"
+layout: spec
+copyrights:
+  -
+    name: "Simon Ser"
+    period: "2024"
+    email: "contact@emersion.fr"
+---
+
+## Notes for implementing work-in-progress version
+
+This is a work-in-progress specification.
+
+Software implementing this work-in-progress specification MUST NOT use the
+unprefixed `extended-isupport` capability name. Instead, implementations SHOULD
+use the `draft/extended-isupport` capability name to be interoperable with
+other software implementing a compatible work-in-progress version. The final
+version of the specification will use unprefixed capability names.
+
+## Introduction
+
+`RPL_ISUPPORT` is used to advertise metadata about the server as key-value
+entries. However, `RPL_ISUPPORT` is only sent by servers after connection
+registration. This undermines the usefulness of `RPL_ISUPPORT`: some of the
+metadata would be useful to clients prior to connection registration. This
+extension adds a mechanism to send `RPL_ISUPPORT` messages before connection
+registration, as well as an end of `RPL_ISUPPORT` list indication.
+
+## Implementation
+
+### `draft/extended-isupport` capability
+
+When the `draft/extended-isupport` capability is enabled by the client, the
+server MUST send one or more `RPL_ISUPPORT` messages (grouped inside a
+`draft/isupport` batch if the `batch` capability is enabled). The capability
+MAY be enabled by the client before connection registration completes (ie,
+before the client sends `CAP END`, and before the server sends `RPL_WELCOME`).
+
+Before connection registration completes, while `draft/extended-isupport` is
+enabled, the server MAY send updates to the key-value entries via subsequent
+`RPL_ISUPPORT` messages (the same way it would after connection registration
+completes without this extension).
+
+The server MAY skip the `RPL_ISUPPORT` replies usually sent when connection
+registration completes, if it already sent all information.
+
+Before connection registration, the server MAY send only a subset of the full
+`RPL_ISUPPORT` list. In that case, the server MUST send a `RPL_ISUPPORT` list
+when connection registration completes with entries previously omitted.
+
+The server MUST send one or more `RPL_ISUPPORT` messages when the capability is
+enabled, including after connection registration or if the client enables,
+disables and then enables again the capability. If the client requests the
+`draft/extended-isupport` capability while it's already enabled, the server
+MUST NOT send any `RPL_ISUPPORT` messages in response.
```

Comment:
```suggestion
The server MUST send one or more `RPL_ISUPPORT` messages when the capability is
enabled, including after connection registration or if the client disables then re-enables
the capability.
```

Tidied wording a bit. I don't think the thing about enabling when already enabled is necessary, it feels like
a) behaviour that should be defined (if at all) in the cap spec
b) unlikely to be something server cap implementations will trip over
c) not a huge problem if you get sent those messages again, if you implement the client correctly.

Reply:
Yeah. I just didn't want clients to rely on this. But I don't mind dropping it.